prow/plugins/cherrypickunapproved/cherrypick-unapproved.go

```diff
@@ -0,0 +1,141 @@
+// `cherry-pick-approved` label.
+package cherrypickunapproved
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "cherry-pick-unapproved"
+
+const (
+	cpUnapprovedLabel = "do-not-merge/cherry-pick-not-approved"
+	cpApprovedLabel   = "cherry-pick-approved"
+)
+
+func init() {
+	plugins.RegisterPullRequestHandler(pluginName, handlePullRequest, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	// Only the 'Config' and Description' fields are necessary because this
+	// plugin does not react to any commands.
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "Label PRs against a release branch which do not have the `cherry-pick-approved` label with the `do-not-merge/cherry-pick-not-approved` label.",
+		Config: map[string]string{
+			"": fmt.Sprintf(
+				"The cherry-pick-unapproved plugin treats PRs against branch names satisfying the regular expression `%s` as cherry-pick PRs and adds the following comment:\n%s",
+				config.CherryPickUnapproved.BranchRegexp,
+				config.CherryPickUnapproved.Comment,
+			),
+		},
+	}
+	return pluginHelp, nil
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	AddLabel(owner, repo string, number int, label string) error
+	RemoveLabel(owner, repo string, number int, label string) error
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+}
+
+type commentPruner interface {
+	PruneComments(shouldPrune func(github.IssueComment) bool)
+}
+
+func handlePullRequest(pc plugins.PluginClient, pr github.PullRequestEvent) error {
+	return handlePR(pc.GitHubClient, pc.Logger, &pr, pc.CommentPruner, pc.PluginConfig.CherryPickUnapproved.BranchRe, pc.PluginConfig.CherryPickUnapproved.Comment)
+}
+
+func handlePR(gc githubClient, log *logrus.Entry, pr *github.PullRequestEvent, cp commentPruner, branchRe *regexp.Regexp, comment string) error {
+	// Only consider the events that indicate opening of the PR and
+	// when the cpApproved and cpUnapproved labels are added or removed
+	cpLabelUpdated := (pr.Action == github.PullRequestActionLabeled || pr.Action == github.PullRequestActionUnlabeled) &&
+		(pr.Label.Name == cpApprovedLabel || pr.Label.Name == cpUnapprovedLabel)
+	if pr.Action != github.PullRequestActionOpened && pr.Action != github.PullRequestActionReopened && !cpLabelUpdated {
+		return nil
+	}
+
+	var (
+		org    = pr.Repo.Owner.Login
+		repo   = pr.Repo.Name
+		branch = pr.PullRequest.Base.Ref
+	)
+
+	// if the branch doesn't match against the branch names allowed for cherry-picks,
+	// don't do anything
+	if !branchRe.MatchString(branch) {
+		return nil
+	}
+
+	labels, err := gc.GetIssueLabels(org, repo, pr.Number)
+	if err != nil {
+		log.WithError(err).Errorf("failed to list labels")
+		return err
```

Comment:
The returned error will be logged so the log message here is redundant.

Reply:
Done.